plugin/emacs/CMakeLists.txt

```diff
@@ -4,13 +4,54 @@
 cmake_minimum_required(VERSION 3.10)
 include(GNUInstallDirs)
 
-install(
-  FILES flycheck-quicklintjs.el
-  FILES lsp-quicklintjs.el
-  FILES eglot-quicklintjs.el
-  FILES flymake-quicklintjs.el
-  DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/emacs/site-lisp"
-)
+find_package(QuickLintJsEmacs)
+
+if (QUICK_LINT_JS_EMACS_FOUND)
+  emacs_pkg_target(emacs-plugin-quicklintjs
+    ${CMAKE_CURRENT_LIST_DIR}/quicklintjs.el 0.0.1
+    OUTPUT quicklintjs-0.0.1)
+
+  emacs_pkg_target(emacs-plugin-flycheck-quicklintjs
+    ${CMAKE_CURRENT_LIST_DIR}/flycheck-quicklintjs.el 0.0.1
+    OUTPUT flycheck-quicklintjs-0.0.1)
+
+  emacs_pkg_target(emacs-plugin-flymake-quicklintjs
+    ${CMAKE_CURRENT_LIST_DIR}/flymake-quicklintjs.el 0.0.1
+    OUTPUT flymake-quicklintjs-0.0.1)
+
+  emacs_pkg_target(emacs-plugin-eglot-quicklintjs
+    ${CMAKE_CURRENT_LIST_DIR}/eglot-quicklintjs.el 0.0.1
+    OUTPUT eglot-quicklintjs-0.0.1)
+
+  emacs_pkg_target(emacs-plugin-lsp-quicklintjs
+    ${CMAKE_CURRENT_LIST_DIR}/lsp-quicklintjs.el 0.0.1
+    OUTPUT lsp-quicklintjs-0.0.1)
+
+  add_dependencies(emacs-plugin-flycheck-quicklintjs
+    emacs-plugin-quicklintjs)
+  add_dependencies(emacs-plugin-flymake-quicklintjs
+    emacs-plugin-quicklintjs)
+  add_dependencies(emacs-plugin-eglot-quicklintjs
+    emacs-plugin-quicklintjs)
+  add_dependencies(emacs-plugin-lsp-quicklintjs
+    emacs-plugin-quicklintjs)
+
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/quicklintjs-0.0.1
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/emacs/site-lisp/elpa)
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/flycheck-quicklintjs-0.0.1
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/emacs/site-lisp/elpa)
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/flymake-quicklintjs-0.0.1
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/emacs/site-lisp/elpa)
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/eglot-quicklintjs-0.0.1
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/emacs/site-lisp/elpa)
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lsp-quicklintjs-0.0.1
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/emacs/site-lisp/elpa)
+endif ()
```

Comment:
> Could we make just one package?

Yes, to use what emacs calls "multi-file" packages, it would require `tar` available while building the packages tho, i didn't use it here because i imagine that's unlikely to `tar` be available on a Windows machine, what you think? i'm fine with both ways.

Reply:
> it would require `tar` available while building the packages tho, i didn't use it here because i imagine that's unlikely to tar be available on a Windows machine, what you think?

CMake provides a tool to create `tar` archives: https://cmake.org/cmake/help/v3.21/manual/cmake.1.html#run-a-command-line-tool

```
$ cmake -E tar czf src.tar.gz src
$ file src.tar.gz
src.tar.gz: gzip compressed data, last modified: Fri Aug 13 18:15:35 2021, from Unix, original size 1392640
```

Reply:
@strager i made the change to use a tar archive as a single package, besides that it feel clunky, it's discouraged by package archives, do you think the last "problem" in this list applies here? i do think. https://github.com/melpa/melpa/blob/5d49574/CONTRIBUTING.org#fixing-typical-problems

Reply:
> it's discouraged by package archives

Having multiple .el files in a package is discouraged? O_o I'm unsure if that's what the documentation actually says.

> do you think the last "problem" in this list applies here? i do think. https://github.com/melpa/melpa/blob/5d49574/CONTRIBUTING.org#fixing-typical-problems

It says:

> MELPA **only** looks at the main .el file (or -pkg.el file, if provided, though we discourage that).

What does it mean to "look at" an .el file?

Reply:
> What does it mean to "look at" an .el file?

In short, an .el file can be package on its own, its format something along the lines:
```
;;; <name>.el ---
;;; <headers>

<code>

;;; <name>.el ends here
```

so when it says "look at" an .el file, is an .el file which is in this package format, when a package has multiple .el files, this format is not used, instead there should be a tar archive that contains all .el files along with a magic `<name>-pkg.el`, which must call `(define-package ...args)`.

Reply:
> [a tar archive is] discouraged by package archives, do you think the last "problem" in this list applies here? i do think. https://github.com/melpa/melpa/blob/5d49574/CONTRIBUTING.org#fixing-typical-problems

I think we should do one of two things:
* merge all of our .el files into one .el file
* create a tar, even though MELPA docs say "we discourage that"

I'm happy with the tar if it's already implemented and if it makes developing the plugin easier for us.